### PR TITLE
chore: simplify builder Dockerfiles

### PR DESCRIPTION
**Pull request by arbourd**
Removes uneeded WORKDIRs. Cleans up order of operations. Mixes some
layers intentionally.

### CODE REVIEW

File: builder/docker/assemblyscript/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,6 @@
...
 FROM node:16-buster-slim
-
-RUN npm install -g npm@latest
-
+WORKDIR /root/runnable
 COPY --from=subo /go/bin/subo /usr/local/bin
-
-WORKDIR /root/runnable
\ No newline at end of file
+RUN npm install -g npm@latest
```
Comment (jagger27) [inline builder/docker/assemblyscript/Dockerfile:6]: Just for my own curiosity, was moving this line below the `COPY` significant for some weird Docker caching/optimization reason?

File: builder/docker/grain/Dockerfile [hunk 2]
```diff
@@ -1,10 +1,6 @@
...
-WORKDIR /root
-RUN mkdir runnable; mkdir suborbital
-
-COPY --from=subo /go/bin/subo /usr/local/bin
-
-WORKDIR /root/runnable
\ No newline at end of file
+WORKDIR /root/runnable
+COPY --from=subo /go/bin/subo /usr/local/bin/subo
+RUN mkdir mkdir /root/suborbital
```
Comment (jagger27) [inline builder/docker/grain/Dockerfile:6]:
```suggestion
RUN mkdir /root/suborbital
```
Extra `mkdir`

**Review by jagger27 — COMMENTED** [review]
I see that we removed `mkdir runnable` everywhere, but we're still using `WORKDIR /root/runnable`. Where is that folder being created now?

File: builder/docker/assemblyscript/Dockerfile (hunk 1 above)
Comment (arbourd) [reply builder/docker/assemblyscript/Dockerfile:6]: In this case it shouldn't really matter, but ya it's always best to have runs at the bottom to maintain prevent cache busting.

**Review by jagger27 — APPROVED** [review]

**Pull request merged**